django/utils/autoreload.py

```diff
@@ -16,7 +16,7 @@
 
```

Comment:
### `sys.gettrace`

Debuggers use [`sys.settrace`](https://docs.python.org/3/library/sys.html#sys.settrace) in their implementation. We could use [`sys.gettrace`](https://docs.python.org/3/library/sys.html#sys.gettrace) to check if there are any active trace functions but settrace is thread specific by default. 

> The function is thread-specific; for a debugger to support multiple threads, it must register a trace function using settrace() for each thread being debugged or use threading.settrace().

I was unable to use gettrace with [`pudb`](https://github.com/inducer/pudb/blob/main/pudb/debugger.py#L276) or `pdb` to detect when a debugger is activate. That said, I am not super familiar with threading and I can't think of a way to check the result of `sys.gettrace()` in another thread in any case.

http://recordit.co/UXXAh3F3YZ

<details> <summary>Diff</summary>

```
diff --git a/django/utils/autoreload.py b/django/utils/autoreload.py
index 3847252632..ec95db64fd 100644
--- a/django/utils/autoreload.py
+++ b/django/utils/autoreload.py
@@ -367,6 +367,7 @@ class StatReloader(BaseReloader):
     def tick(self):
         mtimes = {}
         while True:
+            print(f"StatReloader.tick: {sys.gettrace()=}")
             for filepath, mtime in self.snapshot_files():
                 old_time = mtimes.get(filepath)
                 mtimes[filepath] = mtime
```

</details>